client/ayon_core/pipeline/stagingdir.py

```diff
@@ -0,0 +1,220 @@
+    no staging dir profile is found.
+    If `prefix` or `suffix` is not set, default values will be used.
+
+    Arguments:
+        host_name (str): Name of host.
+        project_entity (Dict[str, Any]): Project entity.
+        folder_entity (Dict[str, Any]): Folder entity.
+        task_entity (Dict[str, Any]): Task entity.
+        product_type (str): Type of product.
+        product_name (str): Name of product.
+        anatomy (ayon_core.pipeline.Anatomy): Anatomy object.
+        project_settings (Optional[Dict[str, Any]]): Prepared project settings.
+        **kwargs: Arbitrary keyword arguments. See below.
+
+    Keyword Arguments:
+        force_temp (bool): If True, staging dir will be created as tempdir.
+        always_return_path (bool): If True, staging dir will be created as
+            tempdir if no staging dir profile is found.
+        prefix (str): Prefix for staging dir.
+        suffix (str): Suffix for staging dir.
+        formatting_data (Dict[str, Any]): Data for formatting staging dir
+            template.
+
+    Returns:
+        Dict[str, Any]: Staging dir data
+    """
+
+    log = kwargs.get("log") or Logger.get_logger("get_staging_dir")
+    always_return_path = kwargs.get("always_return_path")
+
+    # make sure always_return_path is set to true by default
+    if always_return_path is None:
+        always_return_path = True
+
+    if kwargs.get("force_temp"):
+        return get_temp_dir(
+            project_name=project_entity["name"],
+            anatomy=anatomy,
+            prefix=kwargs.get("prefix"),
+            suffix=kwargs.get("suffix"),
+        )
+
+    # making fewer queries to database
+    ctx_data = get_template_data(
+        project_entity, folder_entity, task_entity, host_name
+    )
+    # add roots to ctx_data
+    ctx_data["root"] = anatomy.roots
+
+    # add additional data
+    ctx_data.update({
+        "product": {
+            "type": product_type,
+            "name": product_name
+        },
+        "host": host_name,
+    })
+
+    # add additional data from kwargs
+    if kwargs.get("formatting_data"):
+        ctx_data.update(kwargs.get("formatting_data"))
+
+    # get staging dir config
+    staging_dir_config = get_staging_dir_config(
+        host_name,
+        project_entity["name"],
+        task_entity["type"],
+        task_entity["name"],
+        product_type,
+        product_name,
+        project_settings=project_settings,
+        anatomy=anatomy,
+        log=log,
+    )
+
+    # if no preset matching and always_get_some_dir is set, return tempdir
+    if not staging_dir_config and always_return_path:
+        return {
+            "stagingDir": get_temp_dir(
+                project_name=project_entity["name"],
+                anatomy=anatomy,
+                prefix=kwargs.get("prefix"),
+                suffix=kwargs.get("suffix"),
+            ),
+            "stagingDir_persistent": False,
```

Comment:
Combination of camel case and snake case. I would rename to `stagingDirPersistent` or `staging_dir_persistent`.

Reply:
Yes, it is in pyblish plugins, question is if this function should return the same, in other words, can you use this function elsewhere? If yes, then please use `staging_dir_persistent` and `get_instance_staging_dir` should care about conversion.

Reply:
@jakubjezek001 @iLLiCiTiT what do you guys want me to do with this one ? I'm not 100% aware from the potential impact of this change.

From what I can see it's write that most of those functions are new code and can be used without pyblish.
But those line here make think we might introduce mismatch in publish instances if we just swap the keys:
- legacy keys expected here: https://github.com/ynput/ayon-core/blob/7375538f22d84b8a4493fc84f77cfd4d12cd8fbc/client/ayon_core/plugins/publish/cleanup_farm.py#L39
- new keys being insterted here : https://github.com/ynput/ayon-core/blob/7375538f22d84b8a4493fc84f77cfd4d12cd8fbc/client/ayon_core/pipeline/create/creator_plugins.py#L883